backend/app/models/case.rb

```diff
@@ -93,6 +95,12 @@ def attachment_count
   end
 
   private
+
+    def set_unique_code
+      highest_code = Case.maximum(:code).to_s.sub("CODE-", '').to_i
+      self.code = "CODE-#{highest_code + 1}"
+    end
```

Comment:
**Suggestion:** Consider using a more robust way to generate unique codes. The current method may lead to race conditions where two cases could potentially get the same code. [possible issue]
```suggestion
    def set_unique_code
      self.code = generate_unique_code
    end

    def generate_unique_code
      loop do
        new_code = "CODE-#{SecureRandom.hex(4)}"
        break new_code unless Case.exists?(code: new_code)
      end
    end
```